src/expr/src/scalar/func.rs

```diff
@@ -4200,6 +4208,7 @@ impl UnaryFunc {
             | CastStringToVarChar(_)
             | CastCharToString(_)
             | CastFloat32ToFloat64(_) => unreachable!(),
+            NotImplemented { function_name } => f.write_str(function_name),
```

Comment:
I don't have a good understanding of what this method does, but can you try dropping it into the unreachable block above?